LibertyDSNP/parquetjs
# Reference Tests and Breaking Change: Optional nullable fields are now null instead of undefined

Problem
=======

We wanted to add tests for all the tests in https://github.com/apache/parquet-testing

### Discovered Bugs
- We treated nulls as undefined, but others don't
- We incorrectly processed dictionary_page_offset >= 0 instead of only > 0


Solution
========

- Added new test that automatically tests all files: `test/reference-test/read-all.test.ts`
- Fixed found bugs

with @shannonwells 

Steps to Verify:
----------------
1. Run the tests
1. Comment out the bug fixes and see reference test files fail
